addons/account/models/account_move.py

```diff
@@ -3576,6 +3597,11 @@ def _compute_analytic_tag_ids(self):
                 if rec:
                     record.analytic_tag_ids = rec.analytic_tag_ids
 
+    @api.depends('move_id.payment_reference')
+    def _compute_name(self):
+        for line in self.filtered(lambda l: not l.name and l.account_id.user_type_id.type in ('receivable', 'payable')):
```

Comment:
@rigr-odoo 

all lines should have name set, because the _synchronize_from_moves uses to propagate changes to account.bank.statement.lines

https://github.com/odoo/odoo/blob/b5ca9664ed71f2dbfa6cabc6eeda95143c3f63b6/addons/account/models/account_bank_statement.py#L949


that could lead on an error since payment_ref is a required field

Reply:
Could you provide use a test that fails but shouldn't according to you?

Reply:
what about and opw 3202630, there is video reproducing the issue

also we got an instance were you can reproduce it

Reply:
Yeah, thanks guys

the think was the I have a module doing the same than this, transforming name field into a computed field, so after this commit landed on our instance, we experienced the error.

So for nothing to do here, 

regards!